docs/source/cpp/compute.rst

```diff
@@ -1659,3 +1659,17 @@ replaced, based on the remaining inputs.
   results in a corresponding null in the output.
 
   Also see: :ref:`if_else <cpp-compute-scalar-selections>`.
+
+Random number generation
+~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
Nit: will you move it to `scalar function` section, **above** `vector functions`, see below line
https://github.com/apache/arrow/blob/master/docs/source/cpp/compute.rst#array-wise-vector-functions

We can leave python doc and test as follow up task.

Reply:
Done.